### PR TITLE
Created docs directory, split README contents into separate pages.

### LINKED ISSUE #38: Project README is too large!
## Expected Behavior

ReadMe should provide information about the project but not necessarily contain *all* documentation for the entire project. 

## Current Behavior

ReadMe has all documentation in it and is hard to navigate

## Possible Solution

Set up GitHub pages with Jekyll [just-the-docs](https://pmarsceill.github.io/just-the-docs/) theme and break the documentation into smaller, more manageable pieces. 

- Version used: 3.0.0

### PR DESCRIPTION
**Pull request by icelynjennings**
## Description

Refactors readme. Fixes #38 

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by gargath** [comment]
👍 on the Markdown.

I'm a little unclear as to why we're adding Jeckyll (as opposed to just maintaining .md files in `/docs`), how that is going to run and where the output will then be hosted.
Could someone fill me in?

**Comment by icelynjennings** [comment]
@gargath The Jekyll part picks up the work done for https://github.com/pusher/oauth2_proxy/issues/38. Though the issue does not describe hosting the docs, my guess however is that @JoelSpeed planned to leverage https://pages.github.com in order to host the Jekyll docs, as this is presumably easy for a public repo.

Regardless, I like the idea of raw Markdown just as much.

File: README.md [hunk 1]
```diff
@@ -32,403 +32,9 @@ sha256sum -c sha256sum.txt 2>&1 | grep OK
...
-    add_header Set-Cookie $auth_cookie;
-
-    proxy_pass http://backend/;
-    # or "root /path/to/site;" or "fastcgi_pass ..." etc
-  }
-}
-```
+2.  [Select a Provider and Register an OAuth Application with a Provider](docs/2_auth.md)
+3.  [Configure OAuth2 Proxy using config file, command line options, or environment variables](docs/3_configuration.md)
+4.  [Configure SSL or Deploy behind a SSL endpoint](docs/4_tls.md) (example provided for Nginx)
```
Comment (JoelSpeed) [inline README.md]: I think we should probably add a section called `Documentation` which links to the docs site once it is hosted on GitHub pages

**Comment by JoelSpeed** [comment]
We need to update the documentation to reflect the current master ReadMe and also not delete the existing ReadMe

We will have to merge this in, set up the github pages and do the migration in parts. A PR should follow this one up to add a link to the new documentation site and remove the existing documentation from the ReadMe

**Comment by icelynjennings** [comment]
The readme has been slightly refactored to contain similar information (e.g. wrt configuration) in the same sections.

**Pull request merged**